### PR TITLE
solution

### PR DESCRIPTION
**Pull request by lefteris-L**

### CODE REVIEW

File: src/groupByKey.ts [hunk 1]
```diff
@@ -1,7 +1,24 @@
-type GroupsMap<T> = {
+export interface GroupsMap<T> {
```
Comment (UmizDemud) [inline src/groupByKey.ts:1]: I like to see this style in JS, didn't know we could do that. Reminds me of C langs

**Review by UmizDemud — COMMENTED** [review]
👍 
cool

**Review by UmizDemud — APPROVED** [review]

File: src/groupByKey.ts [hunk 2]
```diff
@@ -1,7 +1,24 @@
...
+      newKeys.push(item[key]);
+    }
+  });
 
-export function groupByKey(items, key) {
-  // write code here;
+  return newKeys.reduce((prev: GroupsMap<Item>, newKey: KeyTypes) => ({
+    ...prev,
+    [newKey]: items.filter((item: Item) => item[key] === newKey),
+  }), {});
```
Comment (meforoma) [inline src/groupByKey.ts:23]: i'm pretty sure you could have it done with just one `.reduce`
try this out and share any concerns.
```
items.reduce((acc, item) => {
    acc[key] = (...acc[key] || []).concat(item)

    return acc;
  }, {})
```

**Review by meforoma — CHANGES_REQUESTED** [review]
please try out a more graceful approach - see the comment

File: src/groupByKey.ts (hunk 2 above)
Comment (lefteris-L) [reply src/groupByKey.ts]: Excellent. Thanks 👍

**Review by meforoma — APPROVED** [review]

**Pull request closed without merging**